devolving_music/lib/song_scores.py

```diff
@@ -12,165 +13,174 @@
 class SongScores():
 
     def __init__(self, event: Event):
-
         self.song_score_dict = ScoreSuite.get_song_scores_dict(event)
-
         self.comparison_submissions = ScoreSuite.get_event_comparisons(event)
 
-    @staticmethod
-    def compare_not_found(
-            comparison_submission: "SongComparison",
-            song1: "ScoreObject",
-            song2: "ScoreObject"):
-
-        return not song1.compare_present(
-            comparison_submission) and not song2.compare_present(comparison_submission)
-
-    @staticmethod
-    def update_song_rating(
-            comparison_submission: "SongComparison",
-            song1: "ScoreObject",
-            song2: "ScoreObject",
-            score_range=30):
-
-        if SongScores.compare_not_found(comparison_submission, song1, song2):
-
-            song1.log_comparison(comparison_submission)
-
-            song2.log_comparison(comparison_submission)
-
-            song1.quality_score, song2.quality_score = elo_rating(
-                song1.quality_score, song2.quality_score, score_range, comparison_submission.first_better)
-
-            song1.energy_score, song2.energy_score = elo_rating(
-                song1.energy_score, song2.energy_score, score_range, comparison_submission.first_peakier)
-
-            song1.post_peak_score, song2.post_peak_score = elo_rating(
-                song1.post_peak_score, song2.post_peak_score, score_range, comparison_submission.first_post_peakier)
-
-
     def get_scores(self):
-
-        # song_submissions song_submissions should be a list of
-
-        # song submissions ordered from least to greatest
-
-        # by songsubmission id
-
+        #calculates scores for all submissions using current comparisons
         for compare in self.comparison_submissions:
-
             song1_index = compare.first_submission.id
-
             song2_index = compare.second_submission.id
-
             song1 = self.song_score_dict[song1_index]
-
             song2 = self.song_score_dict[song2_index]
 
             SongScores.update_song_rating(compare, song1, song2)
 
         return self.song_score_dict
+    
+    def get_compare_submission_random(self, submission_key):
+        key_list = list(self.song_score_dict.keys())
+        key_list.remove(submission_key)
+        # once you have a critical number of comparisons then pull from quality list
+        return random.choice(key_list)
 
-    @staticmethod
-    def get_info_sort(song_submissions_scored: Iterable["SongSubmission"]):
+    def get_compare_submission_linear(self, submission_key):
+        compare_list = self.comparison_submissions
+        if(len(compare_list)==0):
+            return self.get_compare_submission_random(submission_key)
+        first_sub_id=compare_list[-1].first_submission.id
+        if(submission_key==first_sub_id):
+            return self.get_compare_submission_random(submission_key)
```

Comment:
```suggestion
        first_sub_id=compare_list[-1].first_submission.id
        
        if len(compare_list) == 0 or submission_key == first_sub_id):
            return self.get_compare_submission_random(submission_key)
        else:
```

Why not combine the conditions?

Reply:
because you can only set first_sub_id if compare_list is not empty and it's possible for compare_list to be empty

Reply:
oh true